File: backend/dpwh/src/main/java/merchpoint/service/CategoryService.java
package merchpoint.service;

import merchpoint.model.CategoryEntity;
import merchpoint.repository.CategoryRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

@Service
public class CategoryService {
    
    @Autowired
    private CategoryRepository categoryRepository;
    
    // Create
    public CategoryEntity createCategory(CategoryEntity category) {
        return categoryRepository.save(category);
    }
    
    // Read All
    public List<CategoryEntity> getAllCategories() {
        return categoryRepository.findAll();
    }
    
    // Read One
    public Optional<CategoryEntity> getCategoryById(Long categoryId) {
        return categoryRepository.findById(categoryId);
    }
    
    // Update
    public CategoryEntity updateCategory(Long categoryId, CategoryEntity categoryDetails) {
        CategoryEntity category = categoryRepository.findById(categoryId)
                .orElseThrow(() -> new RuntimeException("Category not found with id: " + categoryId));
        
        category.setCategoryName(categoryDetails.getCategoryName());
        category.setDescription(categoryDetails.getDescription());
        
        return categoryRepository.save(category);
    }
    
    // Delete
    public void deleteCategory(Long categoryId) {
        CategoryEntity category = categoryRepository.findById(categoryId)
                .orElseThrow(() -> new RuntimeException("Category not found with id: " + categoryId));
        categoryRepository.delete(category);
    }
}

File: frontend/src/services/categoryService.js
import { api } from './api';

const categoryService = {
  getAllCategories: async () => {
    return await api.get('category');
  },

  getCategoryById: async (id) => {
    return await api.get(`category/${id}`);
  },

  createCategory: async (categoryData) => {
    return await api.post('category', categoryData);
  },

  updateCategory: async (id, categoryData) => {
    return await api.put(`category/${id}`, categoryData);
  },

  deleteCategory: async (id) => {
    return await api.delete(`category/${id}`);
  }
};

export default categoryService;
